qiling/os/windows/structs.py

```diff
@@ -238,6 +269,660 @@ class KUSER_SHARED_DATA(struct.BaseStruct):
+   +0x6090 PackageProcessorSet : _KAFFINITY_EX
+   +0x6138 PackageId        : Uint4B
+   +0x613c PrcbPad86        : Uint4B
+   +0x6140 SharedReadyQueueMask : Uint8B
+   +0x6148 SharedReadyQueue : Ptr64 _KSHARED_READY_QUEUE
+   +0x6150 SharedQueueScanOwner : Uint4B
+   +0x6154 ScanSiblingIndex : Uint4B
+   +0x6158 CoreProcessorSet : Uint8B
+   +0x6160 ScanSiblingMask  : Uint8B
+   +0x6168 LLCMask          : Uint8B
+   +0x6170 CacheProcessorMask : [5] Uint8B
+   +0x6198 ProcessorProfileControlArea : Ptr64 _PROCESSOR_PROFILE_CONTROL_AREA
+   +0x61a0 ProfileEventIndexAddress : Ptr64 Void
+   +0x61a8 DpcWatchdogProfile : Ptr64 Ptr64 Void
+   +0x61b0 DpcWatchdogProfileCurrentEmptyCapture : Ptr64 Ptr64 Void
+   +0x61b8 SchedulerAssist  : Ptr64 Void
+   +0x61c0 SynchCounters    : _SYNCH_COUNTERS
+   +0x6278 PrcbPad94        : Uint8B
+   +0x6280 FsCounters       : _FILESYSTEM_DISK_COUNTERS
+   +0x6290 VendorString     : [13] UChar
+   +0x629d PrcbPad100       : [3] UChar
+   +0x62a0 FeatureBits      : Uint8B
+   +0x62a8 UpdateSignature  : _LARGE_INTEGER
+   +0x62b0 PteBitCache      : Uint8B
+   +0x62b8 PteBitOffset     : Uint4B
+   +0x62bc PrcbPad105       : Uint4B
+   +0x62c0 Context          : Ptr64 _CONTEXT
+   +0x62c8 ContextFlagsInit : Uint4B
+   +0x62cc PrcbPad115       : Uint4B
+   +0x62d0 ExtendedState    : Ptr64 _XSAVE_AREA
+   +0x62d8 IsrStack         : Ptr64 Void
+   +0x62e0 EntropyTimingState : _KENTROPY_TIMING_STATE
+   +0x6430 PrcbPad110       : Uint8B
+   +0x6438 PrcbPad111       : [7] Uint8B
+   +0x6470 AbSelfIoBoostsList : _SINGLE_LIST_ENTRY
+   +0x6478 AbPropagateBoostsList : _SINGLE_LIST_ENTRY
+   +0x6480 AbDpc            : _KDPC
+   +0x64c0 IoIrpStackProfilerCurrent : _IOP_IRP_STACK_PROFILER
+   +0x6514 IoIrpStackProfilerPrevious : _IOP_IRP_STACK_PROFILER
+   +0x6568 SecureFault      : _KSECURE_FAULT_INFORMATION
+   +0x6578 PrcbPad120       : Uint8B
+   +0x6580 LocalSharedReadyQueue : _KSHARED_READY_QUEUE
+   +0x67f0 PrcbPad125       : [2] Uint8B
+   +0x6800 TimerExpirationTraceCount : Uint4B
+   +0x6804 PrcbPad127       : Uint4B
+   +0x6808 TimerExpirationTrace : [16] _KTIMER_EXPIRATION_TRACE
+   +0x6908 PrcbPad128       : [7] Uint8B
+   +0x6940 Mailbox          : Ptr64 _REQUEST_MAILBOX
+   +0x6948 PrcbPad130       : [7] Uint8B
+   +0x6980 SelfmapLockHandle : [4] _KLOCK_QUEUE_HANDLE
+   +0x69e0 PrcbPad135       : [1184] UChar
+   +0x6e80 KernelDirectoryTableBase : Uint8B
+   +0x6e88 RspBaseShadow    : Uint8B
+   +0x6e90 UserRspShadow    : Uint8B
+   +0x6e98 ShadowFlags      : Uint4B
+   +0x6e9c VerwSelector     : Uint2B
+   +0x6e9e PrcbPad139       : Uint2B
+   +0x6ea0 PrcbPad140       : [508] Uint8B
+   +0x7e80 RequestMailbox   : [1] _REQUEST_MAILBOX
+    '''
+
+    # We need a native pointer so that we can write the address of _KTREAD structure
+    native_type = struct.get_native_type(64)
+    pointer_type = native_type
+
+    _fields_ = (
+        ('MxCsr',                       ctypes.c_ulong),
```

Comment:
From the same reason mentioned above (avoiding structures definitions getting affected by the hosting arch), we should use only known size types, like `c_uint32` and `c_uint16` instead of `long` and `short`.